File: Arrays2/gfg_FormLargestNumber.py
from functools import cmp_to_key

class Solution:
    def compare(self, a, b):
        return (1 if str(a) + str(b) > str(b) + str(a) else -1)

    def findLargest(self, arr):
        # Sort using custom comparator
        arr.sort(key=cmp_to_key(self.compare), reverse=True)

        # Edge case: if the largest number is 0, return "0"
        if arr[0] == 0:
            return "0"

        # Concatenate all numbers
        return ''.join(map(str, arr))

File: Arrays2/gfg_FormLargestNumber.cpp
// https://www.geeksforgeeks.org/problems/largest-number-formed-from-an-array1117/1

class Solution {
public:
    static bool compare(int a, int b) {
        string s1 = to_string(a);
        string s2 = to_string(b);
        return s1 + s2 > s2 + s1;
    }

    string findLargest(vector<int>& arr) {
        sort(arr.begin(), arr.end(), compare);

        // Edge case: if the largest number is 0, return "0"
        if (arr[0] == 0) return "0";

        string result = "";
        for (int num : arr) {
            result += to_string(num);
        }
        return result;
    }
};
